weblate/api/serializers.py

```diff
@@ -156,6 +167,24 @@ class Meta:
         )
         extra_kwargs = {"url": {"view_name": "api:role-detail", "lookup_field": "id"}}
 
+    def create(self, validated_data):
+        permissions_validated = validated_data.pop("permissions", [])
+        role = Role.objects.create(**validated_data)
+        for codename in permissions_validated:
+            permission = Permission.objects.get(codename=codename)
+            role.permissions.add(permission)
```

Comment:
Maybe this can be done simpler?

```suggestion
        role.permissions.add(Permission.objects.get(codename__in=permissions_validated))
```

Reply:
I just realized there might be a problem with this. What if someone inputs more than one permission in the permissions array while creating a role. `Permission.objects.get` will throw an error then, right? `.get()` usually expects only one result of query.

Reply:
Yes, filter was supposed to be there. No need to add all though.